### DISCUSSION TITLE
Optimizing async calls

### DISCUSSION THREAD

**Discussion by clayton-leis** [TarekRaafat/autoComplete.js#247] [Q&A]
I'm trying to lighten the load on my API. Say the user types in "i" and the results "pizza", "sushi", and "fresh juice" appear. If they then add a "z", making their query "iz", I don't need to hit my API again to know that the only result should be "pizza".

If someone could just point me in the right direction. Thanks.

**Comment by folknor** [comment]
`data.src` gets the query as written directly in the attached input as a parameter, so before you fire off the request to the API you could do a simple search in the previous results

```js
let prevResults;
new autoComplete({
	data: {
		src: async (query) => {
			// Check prevResults and only trigger an API request if necessary
		},
	},
	resultsList: {
		element: (element, data) => {
			const { matches } = data
			prevResults = matches
		},
	}
})
```
